cmd/api/src/api/v2/saved_queries.go

```diff
@@ -75,12 +78,17 @@ func (s Resources) ListSavedQueries(response http.ResponseWriter, request *http.
 						api.WriteErrorResponse(request.Context(), api.BuildErrorResponse(http.StatusBadRequest, fmt.Sprintf("%s: %s %s", api.ErrorResponseDetailsFilterPredicateNotSupported, filter.Name, filter.Operator), request), response)
 						return
 					}
-
 					queryFilters[name][i].IsStringData = savedQueries.IsString(filter.Name)
 				}
 			}
 		}
 
+		// needed to remove scope query param since we currently
+		// dont support query params that arent filterable
+		maps.DeleteFunc(queryFilters, func(s string, filters model.QueryParameterFilters) bool {
```

Comment:
Move logic to ParseQueryParamFilter function as a carveout, like we do with pagination. This keeps our special logic in one place even if it's similarly hacky.